content/app/commands/goto-file-command.js

```diff
@@ -0,0 +1,130 @@
+define("content/app/commands/goto-file-command", function(require, exports, module) {
+
+	// imports
+	var EventDispatcher = require("content/app/events/event-dispatcher");
+	var Event =require("content/app/events/event");
+	var PopupManager = require("content/app/ui/popup-manager");
+	var GotoFilePopup = require("content/app/ui/popups/goto-file-popup");
+	var AbstractCommand = require("content/app/commands/abstract-command");
+	var CommitPageHelper = require("content/app/github/commit-page-helper");
+	var WindowHelper = require("content/app/github/window-helper");
+
+	/**
+	 * This command is used for navigating used in current visible file.
+	 */
+	var GotoFileCommand = Class(AbstractCommand, {
+
+		$statics : {
+			MARGIN_TOP : 10,
+		},
+
+		/**
+		 * GotoLineCommand constructor.
+		 */
+		constructor : function () {
+			GotoFileCommand.$super.call(this);
+
+			this._popup = null;
+			this._fileWrappers = null;
+
+			this._onSubmitHandler = this._onSubmit.bind(this);
+			this._onCloseHandler = this._onClose.bind(this);
+		},
+
+		/**
+		 * Override. Run command.
+		 */
+		run : function () {
+			this._popup = new GotoFilePopup();
+
+			this._popup.on(Event.SUBMIT, this._onSubmitHandler);
+			this._popup.on(Event.CLOSE, this._onCloseHandler);
+
+			this._fileWrappers = CommitPageHelper.getAllFiles();
+
+			PopupManager.global().show(this._popup);
+		},
+
+		/**
+		 * Override. Cancel command.
+		 */
+		cancel : function () {
+			this._closePopup();
+			this._triggerCancel();
+		},
+
+		/**
+		 * Close popup.
+		 */
+		_closePopup : function () {
+			if (this._popup) {
+				this._popup.off(Event.SUBMIT, this._onSubmitHandler);
+				this._popup.off(Event.CLOSE, this._onCloseHandler);
+
+				PopupManager.global().hide(this._popup);
+
+				this._popup = null;
+			}
+		},
+
+		/**
+		 * Submit event listener on popup.
+		 * @param  {Event} event
+		 */
+		_onSubmit : function (event) {
```

Comment:
:+1: 

Good article and proposal. I really salut the single responsibility principle, since it makes everything easy to understand and maintain. I'll try to stick to it.